### ISSUE TITLE
Local Transformers?

### ISSUE THREAD

**Issue by zakaluka** [Kinto/kinto.js#777]
Hi All,

Was wondering if there is any on-going effort to implement Local Transformers.  The documentation has stated since v1.0.0 (that's as far back as I was able to go) that they are coming soon.  Any information on these would be appreciated.

Thank you.

**Comment by glasserc** [comment]
Hi, like a lot of things in the Kinto world, it's something we're interested in but don't have concrete plans to implement ourselves because we don't have a real use case for them. I don't think they would be difficult to add but they probably wouldn't be much more difficult than just adding a transformation layer that wraps calls to list, get, update, or insert records, which might be an easier solution depending on your use case. Could you describe a little bit what you're working on and why local transformers would be useful?

**Comment by zakaluka** [comment]
The purpose is to encrypt data stored in IndexedDB as the data could contains PII (such as date of birth, address, account information, etc.).  It basically provides piece of mind for the user that, in case they lose their device or are on a shared machine, all information is safe.

**Comment by zakaluka** [comment]
I was thinking that they could be implemented very, very similarly to how Remote Transformers are currently implemented and, as you mentioned, hooked into the IDB's functions.

**Comment by glasserc** [comment]
Yeah, I would be willing to review a patch that added that functionality. Just be careful with that use case because if you store the encryption/decryption keys on disk, then you haven't actually increased security :)

**Comment by zakaluka** [comment]
Agreed!  I will take a stab at it and see how it goes.
